docs/templates.md

```diff
@@ -0,0 +1,352 @@
+
+For example, if we want an object type user with the key being the users email, we would add the following item in the objects array:
+```go
+{
+    "id": "{{ $.email }}",
+    "type": "user"
+}
+```
+Here, `{{ $.email }}` is a key at the root level of the input object (see [above](#transformer-input))
+***
+For an email identity object, we would do somethink like this:
+```go
+{
+    "id": "{{ $.email }}",
+    "type": "identity"
+}
+```
+***
+If we want to link these objects together using a relation, we need to add a new entry in the relations array:
+```go
+{
+    "relation": "identifier",
+    "subject_type": "user",
+    "subject_id": "{{ $.email }}",
+    "object_type": "identity",
+    "object_id": "{{ $.email }}"
+}
+```
+***
+If we want to iterate over a list of items (eg. roles) and add an object for each one:
+```go
+{{ range $i, $element := $.roles }}
+    {{ if $i }},{{ end }}
+    {
+    "id": "{{$element.name}}",
+    "type": "group",
+    "properties":{}
+    }
+{{ end }}
+```
+> note: we need to be careful with the commas. `{{ if $i }},{{ end }}` is handy since it will add a comma before all elements, except the first one.
+***
+Building on this, a basic transform template for this user would look like this:
+
+```go
+{
+  "objects": [
+    {
+      "id": "{{ $.email }}",
+      "type": "user",
+      "displayName": "{{ $.nickname }}",
+      "created_at":"{{ $.created_at }}",
+      "properties":{
+        "email": "{{ $.email }}",
+        "enabled": true,
+        "picture": "{{ $.picture }}"
+        {{ range $key, $value := $.user_metadata }}
+        ,"{{ $key }}": {{ marshal $value }}
+        {{ end }}
+      }
+    },
+    {
+      "id": "{{ $.user_id }}",
+      "type": "identity",
+      "properties": {
+        "verified": true
+      }
+    },
+    {
+        "id": "{{ $.email }}",
+        "type": "identity",
+        "properties": {
+          "verified": {{ .email_verified }}
+        }
+    }
+    {{ if $.roles }}, {{ end }}
+    {{ range $i, $element := $.roles }}
+      {{ if $i }},{{ end }}
+      {
+        "id": "{{$element.name}}",
+        "type": "group",
+        "properties":{}
+      }
+    {{ end }}
+  ],
+  "relations":[
+      {
+        "relation": "identifier",
+        "subject_type": "user",
+        "subject_id": "{{ $.email }}",
+        "object_type": "identity",
+        "object_id": "{{ $.user_id }}"
+      },
+      {
+        "relation": "identifier",
+        "subject_type": "user",
+        "subject_id": "{{ $.email }}",
+        "object_type": "identity",
+        "object_id": "{{ $.email }}"
+      }
+    {{ if $.roles }}, {{ end }}
+    {{ range $i, $element := $.roles }}
+      {{ if $i }},{{ end }}
+      {
+        "relation": "member",
+        "subject_type": "user",
+        "subject_id": "{{$.email}}",
+        "object_type": "group",
+        "object_id": "{{$element.name}}"
```

Comment:
Maybe use the order from https://github.com/aserto-dev/ds-load/pull/104/files#r1450406576